File: ZooZMobileWebJavaLib/src/com/zooz/mobileweb/java/lib/model/Address.java
package com.zooz.mobileweb.java.lib.model;


public class Address {
	
	public enum AddressType {
		billingAddress, shippingAddress
	}
	
	private AddressType addressType;
	private String firstName;
	private String lastName;
	private String phoneCountryCode;
	private String phoneNumber;
	private String street;
	private String city;
	private String state;
	private String country;
	private String zipCode;
	
	
	public Address(AddressType addressType) {
		this.addressType = addressType;
	}


	public AddressType getAddressType() {
		return addressType;
	}


	public void setAddressType(AddressType addressType) {
		this.addressType = addressType;
	}


	public String getFirstName() {
		return firstName;
	}


	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}


	public String getLastName() {
		return lastName;
	}


	public void setLastName(String lastName) {
		this.lastName = lastName;
	}


	public String getPhoneCountryCode() {
		return phoneCountryCode;
	}


	public void setPhoneCountryCode(String phoneCountryCode) {
		this.phoneCountryCode = phoneCountryCode;
	}


	public String getPhoneNumber() {
		return phoneNumber;
	}


	public void setPhoneNumber(String phoneNumber) {
		this.phoneNumber = phoneNumber;
	}


	public String getStreet() {
		return street;
	}


	public void setStreet(String street) {
		this.street = street;
	}


	public String getCity() {
		return city;
	}


	public void setCity(String city) {
		this.city = city;
	}


	public String getState() {
		return state;
	}


	public void setState(String state) {
		this.state = state;
	}


	public String getCountry() {
		return country;
	}


	public void setCountry(String country) {
		this.country = country;
	}


	public String getZipCode() {
		return zipCode;
	}


	public void setZipCode(String zipCode) {
		this.zipCode = zipCode;
	}
	
	public NVPs toNvps() {
		NVPs nvps = new NVPs();
		
		nvps.add(addressType.name() + ".firstName", firstName);
		nvps.add(addressType.name() + ".lastName", lastName);
		nvps.add(addressType.name() + ".phone.countryCode", phoneCountryCode);
		nvps.add(addressType.name() + ".phone.number", phoneNumber);
		nvps.add(addressType.name() + ".street", street);
		nvps.add(addressType.name() + ".city", city);
		nvps.add(addressType.name() + ".state", state);
		nvps.add(addressType.name() + ".country", country);
		nvps.add(addressType.name() + ".zipCode", zipCode);
		
		return nvps;
		
	}
	
	
	
	

}

File: ZooZMobileWebPHPLib/lib/address.php
<?php
	
	require 'lib/nvps.php';

	class Address {
		
		public $addressType;
		public $firstName;
		public $lastName;
		public $phoneCountryCode;
		public $phoneNumber;
		public $street;
		public $city;
		public $state;
		public $country;
		public $zipCode;
		
		function __construct(AddressType $addressType) {
			$this->addressType = $addressType;
		}
		
		function toNVPs() {
			$nvps = new NVPs();
			
			$nvps->add($this->addressType . ".firstName", $this->firstName);
			$nvps->add($this->addressType . ".lastName", $this->lastName);
			$nvps->add($this->addressType . ".phone.countryCode", $this->phoneCountryCode);
			$nvps->add($this->addressType . ".phone.number", $this->phoneNumber);
			$nvps->add($this->addressType . ".street", $this->street);
			$nvps->add($this->addressType . ".city", $this->city);
			$nvps->add($this->addressType . ".state", $this->state);
			$nvps->add($this->addressType . ".country", $this->country);
			$nvps->add($this->addressType . ".zipCode", $this->zipCode);
			
			return $nvps;
		}
		
		
	}
	
	
	class AddressType {
		const billingAddress = "billingAddress";
		const shippingAddress = "shippingAddress";
	}

?>